### DISCUSSION TITLE
Onedrive sync only

### DISCUSSION THREAD

**Discussion by Nexow91** [abraunegg/onedrive#2630] [Q&A] [answered]
Hi I am quite new to the Onedrive Docker. My question would be the following. 

How can I put the data on OnlyOnedrive like on the PC so that no data or storage space is stored locally. 

The problem is I use Onedrive as unlimited storage (backups) and at some point the server is full locally and all data should only be available online, so that no more data is stored locally or no more storage space is used. 

I use Onedrive Docker in Unraid from the Community Appstore. 

Would be good if it was this Onedrive Uploadonly function. 
In the PC app this is called "share storage space" 

Thanks for your answers :)

**Comment by abraunegg** [comment, marked as answer]
@Nexow91 
> How can I put the data on OnlyOnedrive like on the PC so that no data or storage space is stored locally.

When this question is generally asked, the feature that is being sought is 'on-demand' file access. This client does not support this method of access and is called out in the readme and detailed here: https://github.com/abraunegg/onedrive/issues/757

To use this method to access OneDrive, you need to use https://github.com/jstaf/onedriver however I am unsure if there is a Docker instance for this.

However, given the question below about 'upload only' you *can* configure the client to do this (core client functionality) + have the ability to remove the source files locally after uploading (thus .. no space is retained locally, files only exist online). Please *read* the documentation to understand more.

> I use Onedrive Docker in Unraid from the Community Appstore.

The Unraid Community Appstore just provides an interface to this Docker image (which is provided by this repo): https://hub.docker.com/r/driveone/onedrive

There are several issues with using this Docker image on Unraid - primarily how Unraid runs everything as 'root' which is not very secure and 100% causes problems with the Docker container as the user id's are incorrect and file permissions are all messed up (all due to Unraid running things as 'root'). Please take the time to configure this correctly. 

> Would be good if it was this Onedrive Uploadonly function.

As per above, please *read* the documentation - as upload only capability exists:
* https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md
* https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md
* https://github.com/abraunegg/onedrive/blob/master/docs/Podman.md
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-alpha-5/docs/usage.md
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-alpha-5/docs/application-config-options.md
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-alpha-5/docs/docker.md
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-alpha-5/docs/podman.md

**Comment by Nexow91** [comment]
Hi, thank you very much for the quick answer you are not used to on Github my respect for that :) 

Yes, in principle it is always about the local storage space. In the readme there are the variables for OnlyUpload / OnlyDownload so I wasn't sure. 

If I use the "Only Upload" function, do I have to delete the files manually or does the Docker instance do it for me? 

Otherwise, thank you very much for the detailed description. 
Maybe the feature with the encryption on the fly and the on-demand function will come at some point. I would be very happy. 

In the meantime I have set up a VM (Windows) to set up Onedrive via SMB but unfortunately Onedrive does not allow netdrives and no symlinks. So I only have either a Linux VM via smb in onedriver or this Docker instance.

**Reply by abraunegg** [reply]
> If I use the "Only Upload" function, do I have to delete the files manually or does the Docker instance do it for me?

The *best* solution here is to create a Docker 'config' file: https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md#editing-the-running-configuration-and-using-a-config-file

Once you are inspecting the correct volume, create a config file that contains:
```
upload_only = "true"
remove_source_files = "true"
```
This will:
* Upload only
* Remove local files post successful upload


Restart your Docker container using your updated configuration:
```
docker container run -e ONEDRIVE_VERBOSE=1 -v onedrive_conf:/onedrive/conf -v "${ONEDRIVE_DATA_DIR}:/onedrive/data" driveone/onedrive:edge
```

**Reply by abraunegg** [reply]
Please take note of the edit above!

```
remove_source_files = "true"
```

Is needed - not cleanup as previously specified

**Comment by Nexow91** [comment]
Ah everything clear yes then I have to read up on it definitely, because the Unraid Docker is configured differently and I just don't know how to install the custom config or whether it is enough to set the variables in the docker. 

![image](https://github.com/abraunegg/onedrive/assets/73191540/669ec160-9ebe-4351-ac23-75e8e6bbcbef)

![image](https://github.com/abraunegg/onedrive/assets/73191540/ccf9ab78-14cc-4fab-a760-bf8b19d09b1d)

**Reply by abraunegg** [reply]
In that first image `/mnt/user/appdata/onedrive/conf/` create a new file called 'config' and add the relevant 2 lines.
